test/IRGen/generic_metatypes.swift

```diff
@@ -12,13 +12,11 @@
 // RUN: %swift -target x86_64-unknown-linux-gnu -disable-objc-interop -emit-ir -parse-stdlib -primary-file %s | %FileCheck --check-prefix=CHECK --check-prefix=CHECK-64 %s
 
 // REQUIRES: CODEGENERATOR=X86
-
-// FIXME: SR-3591 test is failing after being unintentionally disabled for a while
-// REQUIRES: SR3591
+// REQUIRES: CODEGENERATOR=ARM
```

Comment:
CODEGENERATOR=ARM and CODEGENERATOR=X86 is different aren't mutually exclusive. But splitting up the test is probably still a good idea.